interface.go

```diff
@@ -65,6 +65,23 @@ type Stage interface {
 	CanAuctionPowerPlant(powerPlant PowerPlant) bool
 }
 
+type MoveType uint
+
+const (
+	AuctionMove MoveType = iota
+	BuyResourceMove
+	BuildHomesMove
+	PowerMove
+)
+
+type Move interface {
+	Type() MoveType
+	AuctionParameters() (bool, int)
```

Comment:
These feel more like internal methods of the Move implementations. If you know there are four types of moves, and the only purpose of a move is to be validated, why not just have `Move.IsValid()` that the master game object can use in different contexts? These four different types of moves could hold all logic around move types within the struct implementation